packages/mds-repository/exceptions.ts

```diff
@@ -14,41 +14,23 @@
     limitations under the License.
  */
 
-import logger from '@mds-core/mds-logger'
+import { ConflictError, ServerError } from '@mds-core/mds-utils'
 
 const isError = (error: unknown): error is Error => error instanceof Error
 
-export class RepositoryError extends Error {
-  private static hasProperty<T extends string>(property: T, error: unknown): error is { [P in T]: string } {
-    return error instanceof Error && property in error
-  }
-
-  private static GetProperty<T extends string>(property: T, error: unknown): string | undefined {
-    return (RepositoryError.hasProperty(property, error) && error[property]) || undefined
-  }
+const hasProperty = <T extends string>(property: T, error: unknown): error is { [P in T]: string } =>
+  isError(error) && property in error
 
-  // Type guard to detect repository errors
-  // https://www.postgresql.org/docs/9.2/errcodes-appendix.html
-  static is = {
-    repositoryError: (error: unknown): error is RepositoryError => error instanceof RepositoryError,
-    uniqueViolationError: (error: unknown): error is RepositoryError =>
-      RepositoryError.is.repositoryError(error) && error.code === '23505'
-  }
-
-  static create(error: unknown): RepositoryError {
-    const exception = isError(error)
-      ? new RepositoryError(
-          RepositoryError.GetProperty('detail', error) ?? error.message,
-          RepositoryError.GetProperty('code', error)
-        )
-      : new RepositoryError(typeof error === 'string' ? error : 'Unexpected Error')
-    logger.error(exception)
-    return exception
-  }
+const getProperty = <T extends string>(property: T, error: unknown): string | undefined =>
+  hasProperty(property, error) ? error[property] : undefined
 
-  constructor(message: string, public code?: string) {
-    super(message)
-    Error.captureStackTrace(this, RepositoryError)
-    this.name = `RepositoryError${code ?? ''}`
+export const RepositoryError = (error: unknown) => {
+  if (isError(error)) {
+    const code = getProperty('code', error)
+    if (code === '23505') {
```

Comment:
Looking forward to adding more of these -- FWIW there might be an existing library which maps postgres errors into custom errors... And if there isn't, an opportunity to create one :)